flink-runtime/src/main/java/org/apache/flink/runtime/deployment/CachedShuffleDescriptors.java

```diff
@@ -104,4 +119,30 @@ public void markPartitionFinished(IntermediateResultPartition resultPartition) {
                         checkNotNull(
                                 resultPartitionIdToIndex.get(resultPartition.getPartitionId()))));
     }
+
+    private boolean shouldOffload(ShuffleDescriptorAndIndex[] shuffleDescriptorsToSerialize) {
+        return shuffleDescriptorsToSerialize.length * numConsumers
```

Comment:
I'm wondering why we use `shuffleDescriptorsToSerialize.length` here. It may be a bit confusing for this case:

For a vertex of 10,000 * 10,000 in hybrid shuffle:
1. `shuffleDescriptorsToSerialize` only contains half of the ShuffleDescriptors, it wouldn't be offloaded.
2. Afterwards, `shuffleDescriptorsToSerialize` contains all the ShuffleDescriptors, it would be offloaded.

Could we just use `numConsumers` here?

Reply:
In some scenarios, there may only be a few new shuffleDescriptors to serialize, so it is more efficient to use Akka to transfer them directly.